app/models/robots/plate_to_tube_racks_robot.rb

```diff
@@ -0,0 +1,252 @@
+# frozen_string_literal: true
+
+module Robots
+  # This plate to tube racks robot takes one parent plate, and transfers it to
+  # its child tubes that are on multiple tube racks. The tube racks handled by
+  # this robot are not actual recorded labware. Their barcodes are extracted
+  # from the metadata of the tubes and they are accessed using wrapper objects.
+  # When robot controller calls the robot's verify or perform_transfer actions,
+  # the robot will first initialize its labware store with the plate and tube
+  # rack objects. The plate information comes from the Sequencescape API call,
+  # however the tube rack information comes from the metadata of the downstream
+  # tubes included in the same API response. Therefore, the bed verification of
+  # the tube racks depend on the verification of the plate.
+  #
+  # The destination tube racks are distinguished by their barcodes. We assume
+  # that the tubes on the same tube rack have the same labware purpose. When
+  # multiple tubes of the same purpose and the same position are found, the
+  # latest tube is assumed to be on the tube rack and the other tubes are
+  # ignored. We also assume that there cannot be multiple tube racks with the
+  # tubes of the same labware purpose on the robot at the same time.
+  #
+  # For bed verification, only the etched barcode of the tube racks are scanned,
+  # not the individual tubes. The number of tube racks to be verified not only
+  # depends on the robot's configured relationships but also whether the plate
+  # has children with those purposes.
+  #
+  class PlateToTubeRacksRobot < Robots::SplittingRobot
+    attr_writer :relationships # Hash from robot config into @relationships
+
+    # Option for including downstream tubes and metadata in Plate API response.
+    PLATE_INCLUDES = 'purpose,wells,wells.downstream_tubes,wells.downstream_tubes.custom_metadatum_collection'
+
+    # Returns the well order for getting wells from the plate.
+    #
+    # @return [Symbol] the well order
+    #
+    def well_order
+      :coordinate
+    end
+
+    # Returns the bed class for this robot.
+    #
+    # @return [Class] the bed class
+    def bed_class
+      Robots::Bed::PlateToTubeRacksBed
+    end
+
+    # Performs the transfer between plate and tube racks. This method is called
+    # by the robot controller when the user clicks the start robot button.
+    #
+    # @param [Hash] bed_labwares the bed_labwares hash from request parameters
```

Comment:
```suggestion
    # @param [Hash] bed_labwares the bed_labwares hash from request parameters (from user scanning labware into beds)
```

Reply:
Would also be good to write out the structure of this hash if poss - unless maybe it's detailed elsewhere?